src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,328 @@ class TableLayout
      */
-    ColumnParam( std::string const & name, Alignment align )
-      : columnName( name ), alignment( align )
-    {}
+    CellLayout();
 
     /**
-     * @brief Construct a ColumnParam object with the specified name, alignment, and display flag.
-     * @param name The name of the column
-     * @param align The alignment of the column
-     * @param display Flag indicating whether the column is enabled
+     * @brief Constructor to initialize a cell given celltype, value and alignment.
+     * @param cellType The type of the cell.
+     * @param value The value to be assigned to the cell.
+     * @param alignment The alignment of the cell (left, right, or center).
      */
-    ColumnParam( std::string const & name, Alignment align, bool display )
-      : columnName( name ), alignment( align ), enabled( display )
-    {}
+    CellLayout( CellType cellType, string const & value, TableLayout::Alignment alignment );
   };
 
   /**
-   * @brief Struct for a column.
+   * @class Column
+   * @brief Class representing a column in a table layout.
+   */
+  class Column
+  {
+public:
+    /// The name of the column.
+    CellLayout m_columName;
+    /// A vector containing all sub-columns in the column.
+    std::vector< Column > m_subColumn;
+    /// struct containing m_alignment for the column (header and values)
+    CellAlignment m_cellAlignment;
+    /// Pointer to the parent column (if any).
+    Column * m_parent;
+    /// Pointer to the next column (if any).
+    Column * m_next;
+
+    /// The width of the cell (e.g., for cell containing subColumns).
+    size_t m_cellWidth = 0;
+
+    /**
+     * @brief Default constructor.
+     * Initializes a column with default values.
+     */
+    Column();
+
+    /**
+     * @brief Constructor to initialize a column with a specific `CellLayout`.
+     * @param cell The `CellLayout` object to initialize the column.
+     *
+     */
+    Column( TableLayout::CellLayout cellLayout );
+
+    /**
+     * @brief Sets the name of the column.
+     * @param name The name to set for the column.
+     * @return The current column object.
+     */
+    Column & setName( string_view name );
+
+    /**
+     * @brief Hides the column.
+     * @return The current column objec.
+     */
+    Column & hide();
+
+    /**
+     * @brief Sets the maximum string size for the column.
+     * @param size The size to set as the maximum string length.
+     */
+    void setMaxStringSize( size_t const size );
+
+    /**
+     * @brief Gets the maximum string size of the column.
+     * @return size_t The maximum string size of the column.
+     */
+    size_t getMaxStringSize() const;
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< TableLayout::Column > subCol );
```

Comment:
std::initializer_list<T> doesn't hold references to its elements and wrapper around const

Reply:
Oh, I didn't know, I indeed found that:
> It’s passed by value because it’s cheap. std::initializer_list, being a thin wrapper, is most likely implemented as a pair of pointers, so copying is (almost) as cheap as passing by reference